### ISSUE TITLE
Restic opt-out mode wrongly skips PVs if mounted in completed pods

### ISSUE THREAD

**Issue by thedarkside** [vmware-tanzu/velero#5451]
**What steps did you take and what happened:**
Run backup on a namespace with `--default-volumes-to-restic`. The namespace contains a PVC which is mounted to a running pod and also is being used (NFS: ReadWriteMany!) by periodic jobs which leave completed pods after they ran successfully. 

The contents of the said pvc/pv are **not** being backed up at all!

**What did you expect to happen:**

The contents of the said pvc/pv will get backed up.

**The following information will help us better understand what's going on**:

Ive already cleared up the state, so unfortunately no more information is present. 

But based on the logs ive read it seems that velero blacklists PVs which were mounted on stopped pods ignoring still running pods which mount those PVs too. 
Ive also described what was going on here: https://github.com/vmware-tanzu/velero/issues/4852#issuecomment-1279137634

**Environment:**

- Velero version (use `velero version`): 
```
Client:
	Version: v1.9.2
	Git commit: 82a100981cc66d119cf9b1d121f45c5c9dcf99e1
Server:
	Version: v1.9.2
```
- Velero features (use `velero client config get features`): 
```
features: <NOT SET>
```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.15+k3s1", GitCommit:"7b69bebdfa68ca19dd67df4d59d17570437e86ee", GitTreeState:"clean", BuildDate:"2022-09-28T17:52:22Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.15+k3s1", GitCommit:"7b69bebdfa68ca19dd67df4d59d17570437e86ee", GitTreeState:"clean", BuildDate:"2022-09-28T17:52:22Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes installer & version:
K3s
- Cloud provider or hardware configuration:
Baremetal single node server
- OS (e.g. from `/etc/os-release`):
```
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Comment by qiuming-best** [comment]
@thedarkside it's one bug that the code details is in #5458, we will fix it later

**Comment by Lyndon-Li** [comment]
Besides the question of how to visit this PV during backup, we need to cover more challenges from ReadWriteMany PV:
1. How to prevent the PV from being backed up multiple times (== the number of mounts)?
2. Event if 1 is allowed, we may get different backup data from different mounts, because each backup reads data from the local cache first. So which one do we choose to restore?
3. If we allow the PV to be backed up once only, the backed up data may be even less consistent if we back up data from one node but the write is from the other node, because: 1. the underlying storage usually guarantees the critical metadata (i.e., file system metadata) to be synced among nodes only, the other data(i.e., file data) may not have been there at the time of backup; 2. the underlying storage may do some advanced/intelligent optimizations that the data may not be available from one node at a period of time

**Comment by Lyndon-Li** [comment]
As the first phase, we can support to backup this ReadWriteMany PVs, make sure to backup the PV once and we can ignore the potential data problems for now.

**Comment by Lyndon-Li** [comment]
Fixed by #5769
